pkg/internal/tracking/query_test.go

```diff
@@ -3,12 +3,48 @@ package tracking
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/stretchr/testify/assert"
 	"testing"
 
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
 	"github.com/stretchr/testify/require"
 )
 
+func TestTrimMetadata(t *testing.T) {
+	testCases := []struct {
+		Input    string
+		Expected string
+	}{
+		{
+			Input:    "select 1",
+			Expected: "select 1",
+		},
+		{
+			Input:    "select 1; --some comment",
+			Expected: "select 1; --some comment",
+		},
+		{
+			Input:    fmt.Sprintf("select 1; --%s", MetadataPrefix),
+			Expected: fmt.Sprintf("select 1; --%s", MetadataPrefix),
```

Comment:
I don't get why we expect the query with metadata prefix here. Shouldn't we also trim?

Reply:
I didn't have the space afterwards, but I can make it so that this one is also removed. On the other hand it's not valid because the JSON is not there. I don't think this case would ever be a problem, but I can change it.

Reply:
Done